### ISSUE TITLE
Decompress Chunk Truncated error

### ISSUE THREAD

**Issue by mayurpande** [andrix/python-snappy#99]
I have tried to follow the readme and write this line from my REPL like so:

```
python -m snappy -d temp.snappy temp.txt

```

However I get the error `UncompressError: chunk truncated`

**Comment by mayurpande** [comment]
Also when I try to use it within my script it fails saying:
```
snappy.UncompressError: Error while decompressing: invalid input
```
However I have a bytes like object:

```
    with open('libs/temp.snappy', 'rb') as f:
        data = f.read()
        snappy.uncompress(data)
```

**Comment by martindurant** [comment]
how did you make the file? 
Note the difference between `decompress` and `stream_decompress`.

**Comment by mayurpande** [comment]
It is compressed in S3, then I download it.

**Comment by martindurant** [comment]
> It is compressed in S3

How was is created?

**Comment by mayurpande** [comment]
> > It is compressed in S3
> 
> How was is created?

It is fed through Firehose and then the Firehose handles the compression.

**Comment by martindurant** [comment]
So, this library has three decompress functions, you should try each. Otherwise, you will need to get the detail of what firehose is doing for you. This isn't a parquet file, right?

**Comment by mayurpande** [comment]
> So, this library has three decompress functions, you should try each. Otherwise, you will need to get the detail of what firehose is doing for you. This isn't a parquet file, right?

No, it isn't parquet. As per the information for Firehose it says

> S3 compression and encryption
Kinesis Data Firehose can compress records before delivering them to your S3 bucket. Compressed records can also be encrypted in the S3 bucket using a KMS master key.

So I have tried `uncompress` and `stream_compress` and both do not work,  I will try `decompress` now.

**Comment by martindurant** [comment]
> Kinesis Data Firehose can compress records before delivering them to your S3 bucket.

Sorry, that doesn't give us much to work from. Also, don't forget `hadoop_stream_decompress`.

**Comment by mayurpande** [comment]
> Sorry, that doesn't give us much to work from. Also, don't forget `hadoop_stream_decompress`.

It just means that is compresses into a specified format, with the ability to choose from:

- Disabled
- GZIP
- Snappy
- Zip
- Hadoop-Compatible Snappy

**Comment by mayurpande** [comment]
> I have tried to follow the readme and write this line from my REPL like so:
> 
> ```
> python -m snappy -d temp.snappy temp.txt
> ```
> 
> However I get the error `UncompressError: chunk truncated`

Also there is still the issue of this. Not too sure why all methods are not working, but as have mentioned I am able to use `snzip` from my command line.

**Comment by martindurant** [comment]
> So I have tried uncompress and stream_compress

You meant stream_**de**compress ?
Sounds like that should be the one, guessing from the snzip readme.

**Comment by mayurpande** [comment]
> You meant stream_**de**compress ? Yes, it still doesn't work, throws me 
`snappy.UncompressError: stream missing snappy identifier` I tried to do what they done [here](https://github.com/andrix/python-snappy/issues/40#issuecomment-281192705) but still did not work.
> Sounds like that should be the one, guessing from the snzip readme.
I am just going to use a `subprocess` call to `snzip` as that works. If there is a fix to this, please let me know.

**Comment by martindurant** [comment]
Sorry, I don't have any more suggestions for you. Perhaps someone else does.

**Comment by mayurpande** [comment]
> Sorry, I don't have any more suggestions for you. Perhaps someone else does.

No problem, thank you for the guidance anyways. My colleague was having the same issue with `chunk truncated` as well.

**Comment by mayurpande** [comment]
Anyone with any ideas for this?

**Comment by randomtask2000** [comment]
I have this issue as well. Any feedback is appreciated.

**Comment by martindurant** [comment]
You might want to try the package `cramjam`, which has `cramjam.snappy.decompress` and `cramjam.snappy.decompress_raw` for the framed and unframed formats, respectively. I don't believe it has a CLI, but you could request one.

**Comment by milesgranger** [comment]
> I don't believe it has a CLI, but you could request one.

Just released one, [`pip install cramjam-cli`](https://pypi.org/project/cramjam-cli/) for anyone interested.
